### DISCUSSION TITLE
Question: TAB key direction

### DISCUSSION THREAD

**Discussion by marcotrosi** [skim-rs/skim#881] [General]
Hi @lotabout 
hi everyone reading this,

in your documentation you write ...
TAB | Toggle selection and move down (with -m)
Shift-TAB | Toggle selection and move up (with -m)

So I use -m --reverse and when I press the TAB key that object is marked and the next line will be highlighted.

But to all people that don't want to use --reverse, isn't it annoying that you have to press Shift-TAB to select in the right direction?
Wouldn't it make sense to swap the direction only for non-"--reverse"?
So it's no matter if you use --reverse or not, that you only have to press TAB to mark and the next correct line would be highlighted. Cause I have seen people who don't use --reverse and Shift-TAB either. They use default, then go up a bit so that they can use the TAB to select some objects.

So what am I missing here? There must be a reason why this is like it is today.
Is my thinking wrong? What should be the workflow instead? Shall one really press Shift-TAB?

btw. I am not affected by this because I use --reverse, so for me the TAB key direction is correct.
But I was very confused when I used fzf/sk the first time.

thank you

--marco

**Comment by lotabout** [comment]
> So I use -m --reverse and when I press the TAB key that object is marked and the next line will be highlighted.

Theoretically yes. `TAB` should move **up** in normal mode and **down** in reverse mode by definition.

However I do think it's better to keep the movement direction consistent since we have "muscle memory". That's also the reason why the "up" and "down" arrow keys are consistent in the two modes.

The last thing is whether `TAB` should go up or down. And I believe **down** is the common case.

Comment are welcome :)

**Reply by marcotrosi** [reply]
my muscle memory is that TAB is "next element", and not up or down, and Shift-TAB is previous. And there is not THE ONE muscle memory, so then it should be configurable if there is no common agreement, right?

**Comment by LoricAndre** [comment]
To achieve this behavior, you can use `sk --bind 'tab:toggle-in,btab:toggle-out'` or add it to your SKIM_DEFAULT_OPTIONS (or SKIM_OPTIONS_FILE)
